src/gretel_trainer/benchmark/comparison.py

```diff
@@ -89,14 +112,20 @@ def __init__(
         self.datasets = [
             _make_dataset(dataset, self.config.working_dir) for dataset in datasets
         ]
-        self._gretel_executors: Dict[str, Executor] = {}
-        self._custom_executors: Dict[str, Executor] = {}
+        self._gretel_executors: Dict[Tuple[str, str], Executor] = {}
```

Comment:
Yeah I think this makes sense. Curious, why not a dataclass, then?

Reply:
Good point. I thought it might be just marginally easier to construct this way if you wanted to dive into run-specific details, e.g.:
```
comparison.executors[("GretelAmplify", "iris")]
# vs
comparison.executors[RunIdentifier("GretelAmplify", "iris")]
```
On the other hand / to your point, I think tuples as keys are infrequent enough that the first option above always felt a little odd (in an "oh right, I forgot the parens" way) and so maybe the dataclass approach is best.

Reply:
Changed to a tuple subclass (but named it `RunKey`)